crates/anvil/src/eth/backend/mem/mod.rs

```diff
@@ -1232,7 +1232,15 @@ impl Backend {
             gas_limit: gas_limit as u64,
             gas_price: U256::from(gas_price),
             gas_priority_fee: max_priority_fee_per_gas.map(U256::from),
-            max_fee_per_blob_gas: max_fee_per_blob_gas.map(U256::from),
+            max_fee_per_blob_gas: max_fee_per_blob_gas
+                .or_else(|| {
+                    if blob_versioned_hashes.as_ref().map_or(false, |hashes| !hashes.is_empty()) {
+                        env.block.get_blob_gasprice()
+                    } else {
+                        None
+                    }
+                })
+                .map(U256::from),
```

Comment:
we already have to do `blob_versioned_hashes.unwrap_or_default(),` a few lines below, what we can do here is use a tmp var and check the length, then we don't need the additional as_ref + map_or

Reply:
good call, fixed